yt_dlp/extractor/litv.py

```diff
@@ -68,73 +70,63 @@ def _real_extract(self, url):
-        if self._yes_playlist(series_id, video_id, smuggled_data):
-            playlist_data = self._download_json(
-                'https://www.litv.tv/vod/ajax/getSeriesTree', video_id,
-                query={'seriesId': series_id}, headers={'Accept': 'application/json'})
-            return self._extract_playlist(playlist_data, program_info['contentType'])
-
-        video_data = self._parse_json(self._search_regex(
-            r'uiHlsUrl\s*=\s*testBackendData\(([^;]+)\);',
-            webpage, 'video data', default='{}'), video_id)
-        if not video_data:
-            payload = {'assetId': program_info['assetId']}
-            puid = try_call(lambda: self._get_cookies('https://www.litv.tv/')['PUID'].value)
-            if puid:
-                payload.update({
-                    'type': 'auth',
-                    'puid': puid,
-                })
-                endpoint = 'getUrl'
-            else:
-                payload.update({
-                    'watchDevices': program_info['watchDevices'],
-                    'contentType': program_info['contentType'],
-                })
-                endpoint = 'getMainUrlNoAuth'
-            video_data = self._download_json(
-                f'https://www.litv.tv/vod/ajax/{endpoint}', video_id,
-                data=json.dumps(payload).encode(),
-                headers={'Content-Type': 'application/json'})
-
-        if not video_data.get('fullpath'):
-            error_msg = video_data.get('errorMessage')
-            if error_msg == 'vod.error.outsideregionerror':
+        program_info = traverse_obj(
+            vod_data,
+            ('props', 'pageProps', 'programInformation'),
+            default={}, expected_type=dict)
+        series_id = program_info.get('series_id')
+        playlist_data = traverse_obj(vod_data, ('props', 'pageProps', 'seriesTree'))
+        if playlist_data is not None and self._yes_playlist(series_id, video_id, smuggled_data):
+            content_type = program_info.get('content_type')
+            return self._extract_playlist(playlist_data, content_type)
+
+        asset_id = traverse_obj(program_info, ('assets', 0, 'asset_id'))
+        if asset_id is None:  # live stream case
+            asset_id = program_info.get('content_id')
+            media_type = program_info.get('content_type')
+        else:  # vod case
+            media_type = 'vod'
+        puid = try_call(lambda: self._get_cookies('https://www.litv.tv/')['PUID'].value)
+        if puid is None:
+            puid = str(uuid.uuid4())
+            endpoint = 'get-urls-no-auth'
+        else:
+            endpoint = 'get-urls'
+        payload = {'AssetId': asset_id, 'MediaType': media_type, 'puid': puid}
+        video_data = self._download_json(
+            f'https://www.litv.tv/api/{endpoint}', video_id,
+            data=json.dumps(payload).encode(),
+            headers={'Content-Type': 'application/json'})
+
+        if video_data.get('error'):
```

Comment:
I like to use `traverse_obj` for error handling since the response might not even be a dict
```suggestion
        if error := traverse_obj(video_data, ('error', {dict})):
```